code/weapon/weapons.cpp

```diff
@@ -2382,10 +2382,15 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		ci.dinky = false;
 		required_string("+Armor Type:");
 			stuff_string(fname, F_NAME, NAME_LENGTH);
-		armor_index = armor_type_get_idx(fname);
-		if (armor_index < 0) {
-			Warning(LOCATION, "Armor type '%s' not found for conditional impact in weapon %s!", fname, wip->name);
-		}
+		if (*fname == *"NO ARMOR") {
+			armor_index = -1;
+		} else {
+			armor_index = armor_type_get_idx(fname);
+			if (armor_index < 0) {
+				Warning(LOCATION, "Armor type '%s' not found for conditional impact in weapon %s!", fname, wip->name);
+				continue;
```

Comment:
The `continue` will skip the rest of the parsing.  That will cause future conditional impacts (or even future lines in the weapon entry) to throw parse errors.  Instead, the rest of the parsing should continue as usual, but the entry should not be added to `wip->conditional_impacts`.

Reply:
The `continue` behaves exactly as you linked.  It proceeds to the next iteration, as you said, and it also skips the rest of the parsing, as I said.  But this leaves the entry fields *unparsed*.  That means they will stick around to interfere with whatever is parsed next.

Also, this new change isn't correct.  It won't add the legitimate no-armor case, and it will also warn on both the legitimate and illegitimate no-armor cases.  You were closer the previous time.  What you need to do is check the `armor_index` when first parsed, and warn when first parsed, but then use a flag to skip adding the entry to `wip->conditional_impacts` if it is an illegitimate no-armor case.

Reply:
Oh, okay, I see.